extensions/security/runtime/src/main/java/io/quarkus/security/runtime/interceptor/SecurityConstrainer.java

```diff
@@ -5,18 +5,21 @@
 import javax.inject.Inject;
 import javax.inject.Singleton;
 
-import io.quarkus.security.identity.SecurityIdentity;
+import io.quarkus.runtime.BlockingOperationNotAllowedException;
+import io.quarkus.security.runtime.SecurityIdentityAssociation;
 import io.quarkus.security.spi.runtime.SecurityCheck;
 import io.quarkus.security.spi.runtime.SecurityCheckStorage;
+import io.smallrye.mutiny.Uni;
 
 /**
  * @author Michal Szynkiewicz, michal.l.szynkiewicz@gmail.com
  */
 @Singleton
 public class SecurityConstrainer {
 
+    public static final Uni<Object> CHECK_OK = Uni.createFrom().item(new Object());
     @Inject
-    SecurityIdentity identity;
+    SecurityIdentityAssociation identity;
```

Comment:
`SecurtyIdentity` is injected from `SecurityIdentityAssociation#getIdentity` and the GRPC filter sets it, so why it has to be replaced ?

Reply:
for async invocation, we want to support security checks even if we don't have the entity set yet. That's why the `nonBlockingCheck` uses `SecurityIdentityAssociation#getDeferredIdentity()`.
This is not possible with using `SecurityIdentity` directly

Reply:
Makes sense